exchanges/gateio/gateio_test.go

```diff
@@ -2827,7 +2827,7 @@ const futuresCandlesticksPushData = `{"time": 1678469467, "time_ms": 16784694679
 
 func TestFuturesCandlestickPushData(t *testing.T) {
 	t.Parallel()
-	err := g.WsHandleFuturesData(t.Context(), []byte(futuresCandlesticksPushData), asset.Futures)
+	err := g.WsHandleFuturesData(t.Context(), nil, []byte(futuresCandlesticksPushData), asset.Futures)
```

Comment:
Having `nil` here was my least favourite part of reviewing. I think I've gone over the potential panic scenarios and can conclude that it _shouldn't_ happen, but you've now allowed the possibility to happen.

We all see websocket as highly important functionality though and I have full trust that panics as a result of this change will be addressed with high priority